src/dbus_stats.rs

```diff
@@ -0,0 +1,342 @@
+    pub dbus_broker_peer_accounting: Option<HashMap<String, DBusBrokerPeerAccounting>>,
+    pub dbus_broker_user_accounting: Option<HashMap<u32, DBusBrokerUserAccounting>>,
+}
+
+fn get_u32(dict: &Dict, key: &str) -> Option<u32> {
+    let value_key: Value = key.into();
+    dict.get(&value_key).ok().and_then(|v| match v.flatten() {
+        Some(Value::U32(val)) => Some(*val),
+        _ => None,
+    })
+}
+
+fn get_u32_vec(dict: &Dict, key: &str) ->  Option<Vec<u32>> {
+    let value_key: Value = key.into();
+    dict.get(&value_key).ok().and_then(|v| match v.flatten() {
+        Some(Value::Array(array)) => {
+            let vec: Vec<u32> = array.iter().filter_map(|item| {
+                if let Value::U32(num) = item {
+                    Some(*num)
+                } else {
+                    None
+                }
+            }).collect();
+
+            Some(vec)
+        }
+        _ => None,
+    })
+}
+
+/* Parse DBusBrokerPeerAccounting from OwnedValue.
+ * Expected structure:
+ * struct {
+ *     string ":1.2197907"
+ *     array [
+ *         dict entry(
+ *              string "UnixUserID"
+ *              variant uint32 0
+ *         )
+ *         ... other fields
+ *     ]
+ *     array [
+ *         dict entry(
+ *              string "NameObjects"
+ *              uint32 1
+ *         )
+ *         ... other fields
+ *     ]
+ * }
+ */
+
+fn parse_peer_struct(peer_value: &Value) -> Option<DBusBrokerPeerAccounting> {
+    let peer_struct = match peer_value {
+        Value::Structure(peer_struct) => peer_struct,
+        _ => return None,
+    };
+
+    let fields = peer_struct.fields();
+    if fields.len() < 3 {
+        return None;
+    }
+
+    // 1. Extract name (Str)
+    let name = match &fields[0] {
+        Value::Str(s) => s.to_string(),
+        _ => return None,
+    };
+
+    // 2. Extract credentials dict
+    let credentials = match &fields[1] {
+        Value::Dict(d) => d,
+        _ => return None,
+    };
+
+    // 3. Extract stats dict
+    let stats = match &fields[2] {
+        Value::Dict(d) => d,
+        _ => return None,
+    };
+
+    return Some(DBusBrokerPeerAccounting {
+        name: name.clone(),
+        unix_user_id: get_u32(credentials, "UnixUserID"),
+        process_id: get_u32(credentials, "ProcessID"),
+        unix_group_ids: get_u32_vec(credentials, "UnixGroupIDs"),
+        name_objects: get_u32(stats, "NameObjects"),
+        match_bytes: get_u32(stats, "MatchBytes"),
+        matches: get_u32(stats, "Matches"),
+        reply_objects: get_u32(stats, "ReplyObjects"),
+        incoming_bytes: get_u32(stats, "IncomingBytes"),
+        incoming_fds: get_u32(stats, "IncomingFds"),
+        outgoing_bytes: get_u32(stats, "OutgoingBytes"),
+        outgoing_fds: get_u32(stats, "OutgoingFds"),
+        activation_request_bytes: get_u32(stats, "ActivationRequestBytes"),
+        activation_request_fds: get_u32(stats, "ActivationRequestFds"),
+    });
+}
```

Comment:
```suggestion
   Some(DBusBrokerPeerAccounting {
        name: name.clone(),
        unix_user_id: get_u32(credentials, "UnixUserID"),
        process_id: get_u32(credentials, "ProcessID"),
        unix_group_ids: get_u32_vec(credentials, "UnixGroupIDs"),
        name_objects: get_u32(stats, "NameObjects"),
        match_bytes: get_u32(stats, "MatchBytes"),
        matches: get_u32(stats, "Matches"),
        reply_objects: get_u32(stats, "ReplyObjects"),
        incoming_bytes: get_u32(stats, "IncomingBytes"),
        incoming_fds: get_u32(stats, "IncomingFds"),
        outgoing_bytes: get_u32(stats, "OutgoingBytes"),
        outgoing_fds: get_u32(stats, "OutgoingFds"),
        activation_request_bytes: get_u32(stats, "ActivationRequestBytes"),
        activation_request_fds: get_u32(stats, "ActivationRequestFds"),
    })
}
```

Rust has implicit return in functions